### DISCUSSION TITLE
Can Nunjucks shortcodes do this?

### DISCUSSION THREAD

**Discussion by marcamos** [11ty/eleventy#1687] [Q&A] [answered]
I'm a bit of an eleventy amateur and wonder if Nunjucks shortcodes can do the following…

I'm using eleventy-image in an ever-so-slightly tweaked way where I specify three things in the Nunjucks shortcode (as seen in attached screenshot):
* source image location
* processed images' destination
* alt attribute text

![Screenshot of how I'm using eleventy-image](https://user-images.githubusercontent.com/483840/110987736-8acf0980-833d-11eb-86f0-1f5fc2a3ce11.png)

I'm doing it this way because I want the processed images to live in an `assets/` directory right next to the entry itself.

Anywho, it would be lovely if I didn't have to specify the location in every `image` shortcode instance. Is it possible to use something like `page.url` (or similar) _inside_ those two arguments?

Something _like_:

```
{% image {{ page.url }} + "assets/test-img-1.jpg",
         {{ page.url }} + "assets",
         "Test alt attribute" %}
```

…which definitely doesn't work … but is something _like that_ possible?

**Comment by daflh** [comment, marked as answer]
It's actually similar to Javascript.
```
{% image page.url + "assets/test-img-1.jpg",
         page.url + "assets",
         "Test alt attribute" %}
```

Does it work for you?

**Reply by marcamos** [reply]
Oh wow, I should've tried that — it works! Thank you so much, @dafiulh!
